### PR TITLE
Add tests for permissions of files in archives created by `package`

### PR DESCRIPTION
**Pull request by giordano**
In particular, make sure directories pointed by symlinks don't get borked
read-only permissions, as it happened in Yggdrasil for some time.

This addresses the issue reported in #2185.  I'm not sure this test makes sense
on all operating systems, it can also be restricted to Unix- or Linux-only if
necessary.

### CODE REVIEW

File: test/platformengines.jl [hunk 1]
```diff
@@ -43,6 +43,25 @@ using Test, Pkg.PlatformEngines, Pkg.BinaryPlatforms, SHA
         end
     end
 
+    @testset "Issue #2185" begin
+        probe_platform_engines!()
```
Comment (KristofferC) [inline test/platformengines.jl:47]:
```suggestion
```

Can be deleted, it doesn't do anything

Comment (giordano) [reply test/platformengines.jl]: Oh, wow.  Thanks!

**Comment by KristofferC** [comment]
I'm not sure what part of Pkg this is testing? I'm thinking if this should be in Tar.jl?

**Comment by StefanKarpinski** [comment]
Now that we're always using [Tar.jl](https://github.com/JuliaIO/Tar.jl) for unpacking, it does seem better to test this there.

**Comment by giordano** [comment]
But we were experiencing the issue _before_ the switch to Tar.jl, and the problem was with the generated tarballs, not with their unpacking

**Comment by StefanKarpinski** [comment]
Pkg doesn't control the generation of tarballs though, so why test here?

**Comment by giordano** [comment]
Because it used to, when we had the problem, reason why the reproducer in #2185 had the now useless `probe_platform_engines!()`.

**Comment by StefanKarpinski** [comment]
Can you elaborate? I don't get it.

**Comment by giordano** [comment]
There isn't really much to explain, until a month ago `PlatformEngines.package` was this: https://github.com/JuliaLang/Pkg.jl/blob/4a08f33fd42bed62aee7ee16feaddeef2841245a/src/PlatformEngines.jl#L886-L906 no calls to other julia packages. We've been experienced issues with tarballs generated with wrong permissions since September (see https://github.com/JuliaPackaging/BinaryBuilder.jl/pull/911 and https://github.com/JuliaPackaging/Yggdrasil/issues/1969).

My fault is having opened this PR with 2+ months of delay

**Comment by StefanKarpinski** [comment]
So the regressions we're protecting against with this test are:

1. switching to another method of packing tarballs that reintroduces this problem
2. `Tar.create` introducing the same behavior that `z7` exhibited when unpacking them

The first problem doesn't seem like something that could really happen by accident, whereas the second could (and probably should) be tested in the Tar.jl tests instead. I think that's the point @KristofferC was getting at too.

**Comment by giordano** [comment]
I'm mainly concerned with 1, that's what we've seen.  I agree it's unlikely to see it again in the future, but better safe than sorry?

**Comment by StefanKarpinski** [comment]
🤷🏻‍♂️

**Pull request closed without merging**